core/src/main/java/com/linecorp/armeria/common/util/Holder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.util;
+
+import java.util.function.Supplier;
+
+/**
+ * Mutable object value holder.
+ *
+ * @param <T> Type of value.
+ */
+public final class Holder<T> {
+    private T value;
+
+    private Holder(final T value) {
+        this.value = value;
+    }
+
+    public T getValue() {
+        return value;
+    }
+
+    public void setValue(final T value) {
+        this.value = value;
+    }
+
+    /**
+     * If the value is not {@code null}, return it.
+     * If the value is {@code null}, attempts to compute its value using the given supplier function
+     * and set it into this Holder.
+     *
+     * @param valueSupplier the function to compute a value
+     *
+     * @return the current (existing or computed) value, or null if the computed value is null
+     */
+    public T computeIfAbsent(final Supplier<T> valueSupplier) {
+        if (value == null) {
+            value = valueSupplier.get();
+        }
+        return value;
+    }
+
+    public static <T> Holder<T> of(final T value) {
+        return new Holder<>(value);
+    }
+
+    public static <T> Holder<T> ofEmpty() {
```

Comment:
Could you move the `static` methods before all non-static methods/members?

Reply:
fixed